### PR TITLE
Remove Chisel2 Compatibility mode

### PR DESCRIPTION
**Pull request by cyyself**
This PR is built upon #15 and #19. Bumped all the legacy Chisel2 syntax to Chisel5. And compile successfully on  https://github.com/chipsalliance/chisel/commit/e51fa08c7.

### CODE REVIEW

**Comment by jerryz123** [comment]
Does this still compile correctly for 3.6?

File: src/main/scala/devices/chiplink/Partial.scala [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-import Chisel.{defaultCompileOptions => _, _}
-import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
+import chisel3._ 
+import chisel3.util._
 import freechips.rocketchip.tilelink._
 import freechips.rocketchip.util._
 
 class ParitalExtractor[T <: TLDataChannel](gen: T) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/Partial.scala:10]: Where is the `IO`?

File: src/main/scala/devices/chiplink/Partial.scala [hunk 2]
```diff
@@ -25,38 +25,38 @@ class ParitalExtractor[T <: TLDataChannel](gen: T) extends Module
...
+      state := state - 1.U
+      when (empty)   { state := 8.U }
+      when (io.last) { state := 0.U }
     }
   }
 }
 
 class PartialInjector[T <: TLDataChannel](gen: T) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/Partial.scala:56]: same here.

File: src/main/scala/devices/chiplink/RX.scala [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-import Chisel.{defaultCompileOptions => _, _}
-import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
+import chisel3._ 
+import chisel3.util._
 import freechips.rocketchip.tilelink._
 import freechips.rocketchip.util._
 
 class RX(info: ChipLinkInfo) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/RX.scala:10]: and here

File: src/main/scala/devices/chiplink/SinkA.scala [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 
-import Chisel.{defaultCompileOptions => _, _}
-import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
+import chisel3._ 
+import chisel3.util._
 import freechips.rocketchip.tilelink._
 
 class SinkA(info: ChipLinkInfo) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/SinkA.scala:9]: here

File: src/main/scala/devices/chiplink/SinkB.scala [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 
-import Chisel.{defaultCompileOptions => _, _}
-import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
+import chisel3._
+import chisel3.util._
 import freechips.rocketchip.tilelink._
 
 class SinkB(info: ChipLinkInfo) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/SinkB.scala:9]: .

File: src/main/scala/devices/chiplink/SinkC.scala [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 
-import Chisel.{defaultCompileOptions => _, _}
-import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
+import chisel3._ 
+import chisel3.util._
 import freechips.rocketchip.tilelink._
 
 class SinkC(info: ChipLinkInfo) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/SinkC.scala:9]: .

File: src/main/scala/devices/chiplink/SinkD.scala [hunk 7]
```diff
@@ -1,25 +1,25 @@
...
 
-import Chisel.{defaultCompileOptions => _, _}
-import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
+import chisel3._ 
+import chisel3.util._
 import freechips.rocketchip.tilelink._
 
 class SinkD(info: ChipLinkInfo) extends Module
 {
   val io = new Bundle {
```
Comment (sequencer) [inline src/main/scala/devices/chiplink/SinkD.scala:9]: .

**Review by sequencer — CHANGES_REQUESTED** [review]
use `rg val io = ` to find all unwrapped `IO`.

**Comment by sequencer** [comment]
> Does this still compile correctly for 3.6?

I didn't see any incompatible change in this PR. I prefer renaming it: remove Chisel2 Compatibility mode

But there certainly will be some unconnected signals which can only be found in the elaboration time. Unlike rocket-chip, since there is no CI for this repo, the only way is fixing on the fly...

**Comment by jerryz123** [comment]
> > Does this still compile correctly for 3.6?
> 
> 
> 
> I didn't see any incompatible change in this PR. I prefer renaming it: remove Chisel2 Compatibility mode
> 
> 
> 
> But there certainly will be some unconnected signals which can only be found in the elaboration time. Unlike rocket-chip, since there is no CI for this repo, the only way is fixing on the fly...

I agree. It looks good to me. Just want to check. Also agree fix on the fly is fine for this repo

**Review by jerryz123 — APPROVED** [review]

**Comment by sequencer** [comment]
After compiling successfully on my side. I'll squash and merge this for now. But @cyyself should do the runtime fixing later.

**Comment by sequencer** [comment]
Compiled, Squash and merge this for now, leaving runtime issue to @cyyself

**Pull request merged**